packages/components/src/components/primitives/TextPairing/TextPairing.tsx

```diff
@@ -3,14 +3,20 @@ import { Stack } from '@chakra-ui/layout';
 import { Text } from '../Text';
 import { TextPairingProps, TextPairingStyles } from '.';
 import { useStyleConfig } from '@chakra-ui/system';
+import { useBreakpointValue } from '@chakra-ui/react';
 
 export const TextPairing: FC<TextPairingProps> = ({
   variant,
   label,
   subLabel,
 }) => {
+  const responsiveVariant = variant
+    ? typeof variant === 'string'
+      ? variant
+      : useBreakpointValue(variant)
+    : undefined;
```

Comment:
Could we extract this as a hook?  `useVariantByBreakpoint` ?

Reply:
done!